src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -2560,6 +2563,26 @@ int RGWMetadataHandlerPut_BucketInstance::put_check(const DoutPrefixProvider *dp
     /* existing bucket, keep its placement */
     bci.info.bucket.explicit_placement = old_bci->info.bucket.explicit_placement;
     bci.info.placement_rule = old_bci->info.placement_rule;
+
+    //if the bucket is being deleted, create and store a special log type for
+    //bucket instance cleanup in multisite setup
+    const auto& log = bci.info.layout.logs.back();
+    if (bci.info.bucket_deleted() && log.layout.type != rgw::BucketLogType::Deleted) {
+      const auto index_log = bci.info.layout.logs.back();
+      const int shards_num = rgw::num_shards(index_log.layout.in_index);
+      bci.info.layout.logs.push_back({0, {rgw::BucketLogType::Deleted}});
```

Comment:
the new log gen should probably be `log.gen + 1`

Reply:
yup that was it. fixed!
I have created a testcase testing only secondary zone trimming, because we don't create the new layout on master zone. I think we will need to clean up instance by hand on the master zone or should we automate it in some way?

Reply:
actually, manual clean up on master zone might be a bad idea, because if we forget to clean up, then during the next metadata sync, metadata master zone would end up syncing that bucket instance back to the secondary zones.